### PR TITLE
DEVX-121 add CartDiscountReferenceResolutionUtils

### PR DESCRIPTION
**Pull request by lojzatran**
#### Summary

JIRA: https://commercetools.atlassian.net/browse/DEVX-121

### CODE REVIEW

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                channelResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getDistributionChannel().getId()))
+        .product(
+            productResourceIdentifierBuilder ->
+                productResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getProduct().getId()))
+        .supplyChannel(
+            channelResourceIdentifierBuilder ->
+                channelResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getSupplyChannel().getId()))
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java:139]: `getSupplyChannel()` might be null.

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  }
+
+  private static CartDiscountValueGiftLineItemDraft cloneCartDiscountValueGiftLineItemDraft(
+      final CartDiscountValueGiftLineItem cartDiscountValue) {
+    final CartDiscountValueGiftLineItem cartDiscountValueGiftLineItem = cartDiscountValue;
+    return CartDiscountValueGiftLineItemDraftBuilder.of()
+        .distributionChannel(
+            channelResourceIdentifierBuilder ->
+                channelResourceIdentifierBuilder.id(
+                    cartDiscountValueGiftLineItem.getDistributionChannel().getId()))
```
Comment (salander85) [inline src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java:131]: `getDistributionChannel()` might be null.

**Review by salander85 — CHANGES_REQUESTED** [review]

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java (hunk 1 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java]: https://github.com/commercetools/commercetools-sync-java/pull/1010/commits/1c3670042cc994fd23fe20e4cbea811ab2fa26a1

File: src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java (hunk 2 above)
Comment (lojzatran) [reply src/main/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountReferenceResolutionUtils.java]: https://github.com/commercetools/commercetools-sync-java/pull/1010/commits/1c3670042cc994fd23fe20e4cbea811ab2fa26a1

**Review by salander85 — APPROVED** [review]

**Pull request merged**